adapters/yieldmo/yieldmo.go

```diff
@@ -0,0 +1,141 @@
+package yieldmo
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type YieldmoAdapter struct {
+	endpoint string
+}
+
+type Ext struct {
+	PlacementId string `json:"placement_id"`
+}
+
+func (a *YieldmoAdapter) MakeRequests(request *openrtb.BidRequest) ([]*adapters.RequestData, []error) {
+	var errs []error
+	var adapterRequests []*adapters.RequestData
+
+	adapterReq, errors := a.makeRequest(request)
+	if adapterReq != nil {
+		adapterRequests = append(adapterRequests, adapterReq)
+	}
+	errs = append(errs, errors...)
+
+	return adapterRequests, errors
+}
+
+func (a *YieldmoAdapter) makeRequest(request *openrtb.BidRequest) (*adapters.RequestData, []error) {
+	var errs []error
+
+	// Make a copy as we don't want to change the original request
+	reqCopy := *request
```

Comment:
Please note that this only makes a shallow copy. Anything that is a pointer in `request` will still point to the original datastructure. `[]Imp` is technically a pointer. However PBS makes shallow copies of every `Imp` in the request for every adaptor (which requires a shallow copy of `request` itself, so it can point to the copied slice) so the request you get is already unique to your adapter. You only need to make a copy if you plan on changing the data, and afterward want to access the original values.

Reply:
Thanks, I've removed the unnecessary copying.